### ISSUE TITLE
wouldn't switching to touch break mouse events on touchscreen laptops?

### ISSUE THREAD

**Issue by jedwards1211** [LouisBrunner/react-dnd-multi-backend#4]
Or do laptops with touch just emulate mouse events?
To be safe, I'd rather have the HTML and touch backends active at the same time...I've had both active together via my https://github.com/jcoreio/react-dnd-multi-backend I hastily wrote a long time ago, and never saw any problems.  Though I don't have a touch laptop to test it on

**Comment by LouisBrunner** [comment]
Hey!

Wow, I had no idea you already used this name. When I wanted to use both HTML5 and Touch backends, I searched for a "switch backend" but finally settled on "multi backend" without searching first, sorry about that...

I don't own a touch laptop either, but you can mock the behavior by opening Chrome developer tools (probably works on your browser of choice) and toggling the "device toolbar" (phone+tablet icon, ⌘+ shift + M on macOS). It will change the viewport to mock a mobile device and emit touch events only. If you disable it, you will have mouse events again.

The TouchBackend has an option `enableMouseEvents` that also listen for `mousedown`, etc when enabled. So the way I did it in my Backend is to use HTML5 by default and switch to Touch when detecting a `touchstart` event, but because `enableMouseEvents` is enabled, the `TouchBackend` will work with any device (touch and mouse).

*TL;DR*: it should work with touch laptops because of TouchBackend `enableMouseEvents` and I tested it with Chrome developer tools (not sure if it's how touch laptops work).

Concerning your implementation, I can test it, but I personally had problems while enabling two backends at the same time (conflicting events or something).

**Comment by jedwards1211** [comment]
Hmmm, `react-dnd-touch-backend`'s docs say that mouse support is buggy due to difference in event propagation between mouse and touch events.

**Comment by LouisBrunner** [comment]
You are right, this package doesn't really add much if you are fine with `TouchBackend`. It makes the experience a bit more unified by providing a `Preview` component which is enabled when the current backend doesn't support Drag'n'Drop "ghost" (like the `TouchBackend`).

Because I use SVG, I have been using the `TouchBackend` alone for a year and I never had any problem, so I don't know what they mean by buggy but it seems fine.

Yeah, there are a lot of different backends (e.g. `react-dnd-mouse-backend`), it really depends on what you need exactly. :)

**Comment by jedwards1211** [comment]
Yeah I just started trying TouchBackend alone and haven't seen any problems either, in fact it's better than the HTML5 backend because it doesn't suffer [this issue](https://github.com/react-dnd/react-dnd/issues/735) I just filed.
